Src/ILGPU/StrideTypes.tt

```diff
@@ -437,13 +430,56 @@ namespace ILGPU
             /// <param name="elementIndex">The linear element index.</param>
             /// <returns>The reconstructed index.</returns>
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public readonly <#= indexName #> ReconstructFromElementIndex(
-                int elementIndex)
+            public <#= indexName #> ReconstructFromElementIndex(int elementIndex)
             {
+<#      if (def.Dimension == 1) { #>
+                return new <#= indexName #>(elementIndex);
+<#      } else if (def.Dimension == 2) { #>
+                bool prefer1st =
+                    <#= IndexDimensions[0] #>Stride >
+                    <#= IndexDimensions[1] #>Stride;
+                var index = prefer1st
+                    ? new DenseY(<#= IndexDimensions[0] #>Stride)
+                        .ReconstructFromElementIndex(elementIndex)
+                    : new DenseX(<#= IndexDimensions[1] #>Stride)
+                        .ReconstructFromElementIndex(elementIndex);
+                int minDimension = Utilities.Select(
+                    prefer1st,
+                    <#= IndexDimensions[1] #>Stride,
+                    <#= IndexDimensions[0] #>Stride);
+                return new <#= indexName #>(index.X, index.Y / minDimension);
```

Comment:
QUESTION: Is there a reason why the 2-dimensonal implementation is different between this and lines 496 to 512?

Reply:
Nice catch! The implementations should have been the same.